### DISCUSSION TITLE
changes to building a case, be advised: ctsm's nuopc and cmake

### DISCUSSION THREAD

**Discussion by rgknox** [NGEET/fates#960] [Host Land Model notices]
Both CTSM and E3SM have recently been going through some changes as to how the models are built and how cases are created.

------------

CMAKE:

Both models, since they both use CIME, now make use of "cmake" to build the model. More on cmake can be found here: https://cmake.org/
Here is a thread where they have been tracking issues with E3SM: https://github.com/E3SM-Project/E3SM/issues/3919

If you use a supported machine, it is likely that you are not affected by these changes. But if you maintain your own machine, or your machine has a configuration maintained by CIME but is not tested regularly, you should know the following:

Previously, there were two really important config files that did most of the configuring for your maching: config_machines.xml and config_compilers.xml.  **Now, each machine and/or machine compiler also needs its own cmake configuration file.**  Here are the folders with the configuration files, which you can follow as a template for your machine:

For E3SM: https://github.com/E3SM-Project/E3SM/tree/master/cime_config/machines/cmake_macros
For CTSM: https://github.com/ESMCI/ccs_config_cesm/tree/main/machines/cmake_macros

----------

NUOPC  (which deprecates MCT):

CTSM has a readme here: https://github.com/ESCOMP/CTSM/blob/master/README.NUOPC_driver.md

There was also a discussion that was recorded, here:
https://drive.google.com/file/d/1NtkxyvuOH-dX1SAOU4p-DfMMMKWKuQEG/view

If you **just want your old/current scripts to work**, you can add the following flag to your execution of "./create_case" :

> ./create_case --driver=mct

**Comment by glemieux** [comment]
Thanks @rgknox! 

For quick reference, the update to make NUOPC default came in with CTSM tag [ctsm5.1.dev062](https://github.com/ESCOMP/CTSM/releases/tag/ctsm5.1.dev062).  The upcoming FATES API 18 update will utilize the NUOPC update.  As a reminder, please see this wiki page for determine which fates API is compatible with which host land model tag: https://github.com/NGEET/fates/wiki/Table-of-FATES-API-and-HLM-STATUS.  Or see the FATES User's Guide reference section: https://fates-users-guide.readthedocs.io/en/latest/user/Table-of-FATES-API-and-HLM-STATUS.html

**Comment by glemieux** [comment]
Update: As of ctsm5.1.dev075, the [cime external was updated](https://github.com/ESCOMP/CTSM/blob/ctsm5.1.dev075/doc/ChangeLog) to cime6.0.13 which itself [moved all the cime config files out to a new external called ccs_config_cesm](https://github.com/ESMCI/cime/blob/cime6.0.13/ChangeLog) which contains the cmake_macros directory.  The link in the original issue above has been updated to https://github.com/ESMCI/ccs_config_cesm/tree/main/machines/cmake_macros.

**Comment by glemieux** [comment]
New update: specifying a set of `nthreads` for E3SM cases is necessary for unsupported machines now as this pr stripped out setting these by default: https://github.com/E3SM-Project/E3SM/pull/4928
